executor/write.go

```diff
@@ -880,21 +896,14 @@ func (e *InsertExec) exec(goCtx goctx.Context, rows [][]types.Datum) (Row, error
 		h, err := e.Table.AddRecord(e.ctx, row, false)
 		txn.DelOption(kv.PresumeKeyNotExists)
 		if err == nil {
-			if !sessVars.ImportingData {
+			if !(sessVars.ImportingData || sessVars.StmtCtx.BatchCheck) {
```

Comment:
Why ignore the dirty db when using BatchCheck?

Reply:
We've already checked duplicate rows in batch, it is not necessary to add uncommited rows to dirty db.

Reply:
We've already checked duplicate rows in batch, it is not necessary to add uncommited rows to dirty db. @shenli

Reply:
I should have found this when reviewing.